src/box/lua/config/configdata.lua

```diff
@@ -731,12 +731,32 @@ local function validate_anon(found, peers, failover, leader)
     end
 end
 
+local function validate_misplacing(cconfig)
+    for group_name, group_cfg in pairs(cconfig.groups) do
+        if group_cfg.replicasets == nil or
+                next(group_cfg.replicasets) == nil then
+            error(('group %q should include at ' ..
+                   'least one replicaset.'):format(group_name))
```

Comment:
Nit: Please, use `error(<...>, 0)` in config's code. It would be consistent with other errors from this subsystem.

All the other errors are reported this way (on startup):

```
LuajitError: You are doing X wrong, let's do Y
fatal error, exiting the event loop
```

But if we miss the second `error()`'s argument it looks like so.

```
LuajitError: internal/config/somemodule.lua:237: You are doing X wrong, let's do Y
fatal error, exiting the event loop
```

The report looks like a problem in tarantool's internal code, not like a problem in user's data. I don't want these reports look like something is wrong in tarantool itself.

The same idea is applicable to `config:info().alerts` if an error occurs on reload.

(I understand that file:line may be a useful piece of information for debugging, but I take this cost and trying to write nice error messages to make everything clean for a user.)

Reply:
That's a fair point, fixed.